python_scripts/cgc_bowtie2_build.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2016--, Evguenia Kopylova, Jad Kanbar
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+"""
+Build Bowtie2 database on all reference genomes in Kraken report.
+"""
+
+import click
+import collections
+
+def load_kraken_mpa_report(kraken_mpa_report_fp,
+                           taxa_levels,
+                           read_per_taxa):
+    """Absolute abundance of number of reads matching a defined taxa level.
+    Parameters
+    ----------
+    kraken_mpa_report_fp: str
+        filepath to output of "kraken mpa report"
+    taxa_levels: list
+        list of two elements that includes the taxonomic rank at which
+        to generate summary and rank below to split by
+    read_per_taxa: int
+        integer of number of minimum number of reads to keep per taxa
+
+    Returns
+    -------
+    taxonomies: set
+        set of taxonomies from kraken_mpa_report_fp
+    """
+    taxonomic_abundances= []
+    for report_fp in kraken_mpa_report_fp:
+        with open(report_fp) as report_fp:
+            for line in report_fp:
```

Comment:
would it be possible to break this parser out? seems like a logical point for decomposition. i'm not seeing a paired unit test or set of tests either...?